pkg/quota/install/registry.go

```diff
@@ -17,15 +17,42 @@ limitations under the License.
 package install
 
 import (
-	"k8s.io/client-go/informers"
-	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/kubernetes/pkg/quota"
 	"k8s.io/kubernetes/pkg/quota/evaluator/core"
+	"k8s.io/kubernetes/pkg/quota/generic"
 )
 
-// NewRegistry returns a registry of quota evaluators.
-// If a shared informer factory is provided, it is used by evaluators rather than performing direct queries.
-func NewRegistry(kubeClient clientset.Interface, f informers.SharedInformerFactory) quota.Registry {
-	// TODO: when quota supports resources in other api groups, we will need to merge
-	return core.NewRegistry(kubeClient, f)
+// NewQuotaConfigurationForAdmission returns a quota configuration for admission control.
+func NewQuotaConfigurationForAdmission() quota.Configuration {
+	evaluators := core.NewEvaluators(nil)
+	return generic.NewConfiguration(evaluators, DefaultIgnoredResources())
+}
+
+// NewQuotaConfigurationForControllers returns a quota configuration for controllers.
+func NewQuotaConfigurationForControllers(f quota.ListerForResourceFunc) quota.Configuration {
+	evaluators := core.NewEvaluators(f)
+	return generic.NewConfiguration(evaluators, DefaultIgnoredResources())
+}
+
+// DefaultIgnoredResources returns the default set of resources that quota system
+// should ignore. This is exposed so downstream integrators can have access to them.
+func DefaultIgnoredResources() map[schema.GroupResource]struct{} {
```

Comment:
To re-use this to write a custom quota replenishment controller you'll want to expose `var DefaultIgnoredResource func() map[string]struct{} = defaultIngoredResources`

Reply:
i am not sure i follow this... i had assumed that other callers would have their own mechanisms to create a quota.Configuration

Reply:
Another approach to consider would be to change the signature of NewQuotaConfigurationForAdmission and NewQuotaConfigurationForControllers to receive ignored resources as an argument (and update callers to pass DefaultIgnoredResources as appropriate). Is there any project wide position on package variable reassignment for configuration? I realize the practice is widespread (even idiomatic?) in Go, but it's certainly a controversial topic...

Reply:
@ironcladlou -- i wasn't sure if we will end up adding an option similar to gc for configuring ignored resources either.